AndroidNativeKotlinTemplate/app/src/main/java/com/salesforce/androidnativekotlintemplate/MainActivity.kt

```diff
@@ -184,12 +204,61 @@ class MainActivity : SalesforceActivity() {
 
     // region QR Code Login Via Salesforce Identity API UI Bridge Public Implementation
 
-    private fun useQrCodeLoginUrl() {
-        val uiBridgeApiParameters = uiBridgeApiParametersFromQrCodeLoginUrl(intent.data.toString())
-        SalesforceSDKManager.getInstance().apply {
-            frontDoorBridgeUrl = uiBridgeApiParameters?.frontdoorBridgeUrl
-            frontDoorBridgeCodeVerifier = uiBridgeApiParameters?.pkceCodeVerifier
+    /**
+     * Validates and uses the intent's QR code log in URL.
+     * @param url The QR code log in URL
+     */
+    private fun useQrCodeLogInUrl(url: Uri) {
```

Comment:
This method is now the app's single path for QR Code Log In from external deep links _and_ the internal QR code scanner.  Half that logic used to live in the activity with the QR code scanner.